RIRImageResizing/Classes/NSObject/RIRResizeImageOperation.m

```diff
@@ -0,0 +1,183 @@
+//
+//  RIRResizeImageOperation.m
+//  Pods
+//
+//  Created by Richard Reitzfeld on 11/21/16.
+//
+//
+
+#import "RIRResizeImageOperation.h"
+
+#import "RUConditionalReturn.h"
+
+
+
+
+
+@interface RIRResizeImageOperation ()
+
+#pragma mark - scale
+/**
+ Returns a new `UIImage` instance, which is a resized version of the receiver.
+ 
+ @param newSize    The size that the new UIImage instance aims for. This is not necessarily the size of the final image, as it depends on the target image's size, and the `resizeMode` parameter.
+ @param resizeMode Indicates which type of resize method to use.
+ 
+ @return A new `UIImage` instance, which is a resized version of the receiver.
+ */
+-(nullable UIImage*)rir_scaleToSize:(CGSize)newSize usingMode:(UIImage_RIRResizing_ResizeMode)resizeMode;
```

Comment:
Since all these methods are on a class we're making, we don't need the `rir_` prefix.